src/readable_compacts_transform_helper.js

```diff
@@ -1,117 +1,273 @@
-                'name': packageName
-            }))[0].messages;
-        }
-        if (this.msgsFormat == undefined){
-            throw new Error('json meta format does not match with provided package name');
-        }
-        let msg = this.msgsFormat.filter(itemFilter({
-            'name': msgName
-        }))[0];
-        if (msg ==undefined){
-            throw new Error('Message:' + msgName + ' does not exist!');
-        }
-        this.msgName = msgName;
-    }
-    compactsToReadableString(compactComponents) {
-        if (compactComponents.length == 0) {
-            return '';
-        }
-        return doCompactsReadableMapping(this.msgsFormat, this.msgName, compactComponents);
-    }
-    readableStringToCompacts(readableString) {
-        if (readableString == '') {
-            return [];
-        }
-        return doReadableCompactsMapping(this.msgsFormat, this.msgName, readableString);
-    }
-}
\ No newline at end of file
+import Long from 'long';
+import ByteBuffer from 'bytebuffer';
+
+const findEndBracket = (str) => {
+    let index = str.indexOf(']');
+    while (str[index - 1] == '\\') {
+        index = str.indexOf(']', index + 1);
+    }
+    return index;
+};
+
+const itemFilter = (conditions) => {
+    return (item) => {
+        if (conditions['id'] != undefined) {
+            return item['id'] == conditions['id'];
+        } else if (conditions['name'] != undefined) {
+            return item['name'] == conditions['name'];
+        }
+        return false;
+    };
+};
+
+const getField = (conditions, msgName, mappings) => {
+    let fields = mappings.get(msgName);
+    if (fields == undefined) {
+        throw new Error('No ' + msgName + ' definition:');
+    }
+    let field = fields.filter(itemFilter(conditions))[0];
+    if (field == undefined) {
+        throw new Error('Message ' + msgName + ' does not has corresponding field with conditions:' + JSON.stringify(conditions));
+    }
+    return {
+        'id': field['id'],
+        'type': field['type'],
+        'rule': field['rule'],
+        'name': field['name'],
+        'keytype':field['keytype'],
+    };
+};
+
+const getQualifiedMsgName = (msgName, pkgName) => {
+    if (msgName.indexOf('.') == -1) {
+        if (pkgName != '') {
+            msgName = pkgName + '.' + msgName;
+        }
+    }
+    return msgName;
+};
+
+const comToLong = (com) =>{
+    return new Long(com, 0 , true);
+};
+
+const hextoString = (hex) => {
+    let bf = ByteBuffer.fromHex(hex);
+    return bf.readString(bf.limit);
+};
+
+const first16ComsToString = (coms, transform)=>{
+    let result ='';
+    for (let ii = 0; ii <16 ; ii++){
+        let component = coms.shift();
+        if (component.toInt() != 0){
+            result = result + transform(component);                       }
+    }
+    return result;
+};
+
+const doCompactsReadableMapping = (messageFieldsMapping, msgName, pkgName, compactComponents) => {
```

Comment:
In the go implementation we call the two different variations `literal` and `compact`. Could you replace "readable" with "literal" in the js code as well?

Reply:
"ReadableName" is borrowed from go precise code,  if "literal" (better than "readable ") is the choice, it is fine